File: scala/src/main/scala/flatMap.scala
import com.jambit.practicalresult.*

@main def flatMap: Unit =
  val flatMapped: Either[String, List[Int]] =
    Right(4).flatMap(listService)
            .flatMap(list => if (list.isEmpty) Left("Empty list")
                             else Right(list))

  println(flatMapped)

File: kotlin/src/main/java/FlatMap.kt
import arrow.core.Either
import arrow.core.flatMap

fun main() {
    val flatMapped: Either<String, List<Int>> = Either
        .Right(4)
        .flatMap(listService)
        .flatMap { list ->
            if (list.isEmpty()) Either.Left("Empty list")
            else Either.Right(list)
        }

    println(flatMapped)
}